src/code/z_demo.c

```diff
@@ -1,99 +1,1634 @@
+
+    Math_SmoothStepToS(&actor->world.rot.y, Math_Vec3f_Yaw(&start, &end), 10, 1000, 1);
+    actor->shape.rot.y = actor->world.rot.y;
+}
+
+/**
+ * Interpolates the actor's XZ position and yaw based on the corresponding actor action's
+ * position and the current cutscene frame
+ */
+void Cutscene_ActorTranslateXZAndYawSmooth(Actor* actor, GlobalContext* globalCtx, s32 actorActionIndex) {
+    Vec3f start;
+    Vec3f end;
+    CsCmdActorAction* entry;
+    f32 progress;
+
+    start.x = globalCtx->csCtx.actorActions[actorActionIndex]->startPos.x;
+    start.z = globalCtx->csCtx.actorActions[actorActionIndex]->startPos.z;
+    end.x = globalCtx->csCtx.actorActions[actorActionIndex]->endPos.x;
+    end.z = globalCtx->csCtx.actorActions[actorActionIndex]->endPos.z;
+
+    entry = globalCtx->csCtx.actorActions[actorActionIndex];
+    progress = Environment_LerpWeight(entry->endFrame, entry->startFrame, globalCtx->csCtx.frames);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/nop_800EDDB0.s")
+    actor->world.pos.x = start.x + (end.x - start.x) * progress;
+    actor->world.pos.z = start.z + (end.z - start.z) * progress;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDDBC.s")
+    Math_SmoothStepToS(&actor->world.rot.y, Math_Vec3f_Yaw(&start, &end), 10, 1000, 1);
+    actor->shape.rot.y = actor->world.rot.y;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDDCC.s")
+s32 Cutscene_GetSceneSetupIndex(GlobalContext* globalCtx) {
+    s32 sceneSetupIndex = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDE34.s")
+    if (gSaveContext.sceneSetupIndex > 0) {
+        sceneSetupIndex = gSaveContext.sceneSetupIndex;
+    }
+    return sceneSetupIndex;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDF24.s")
+s32 Cutscene_GetActorActionIndex(GlobalContext* globalCtx, u16 actorActionCmd) {
+    s32 i;
+    s32 index = -1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDF78.s")
+    for (i = 0; i < ARRAY_COUNT(D_801F4DC8); i++) {
+        if (actorActionCmd == D_801F4DC8[i]) {
+            index = i;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EE0CC.s")
+    return index;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EE1D8.s")
+s32 Cutscene_CheckActorAction(GlobalContext* globalCtx, u16 actorActionCmd) {
+    if (globalCtx->csCtx.state != CS_STATE_0) {
+        s32 index = Cutscene_GetActorActionIndex(globalCtx, actorActionCmd);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EE200.s")
+        if (index != -1) {
+            return globalCtx->csCtx.actorActions[index] != NULL;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EE29C.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EE2F4.s")
+u8 Cutscene_IsPlayingCs(GlobalContext* globalCtx) {
+    return gSaveContext.cutsceneTrigger != 0 || globalCtx->csCtx.state != CS_STATE_0;
+}
```

Comment:
```suggestion
u8 Cutscene_IsPlaying(GlobalContext* globalCtx) {
    return gSaveContext.cutsceneTrigger != 0 || globalCtx->csCtx.state != CS_STATE_0;
}
```
I wonder if the second `Cs` is redundant. Thoughts?